dashboard/ltiv1p3.py

```diff
@@ -0,0 +1,144 @@
+    email = launch_data['email']
+    first_name = launch_data['given_name']
+    last_name = launch_data['family_name']
+    user_img = launch_data['picture']
+    RANDOM_PASSWORD_DEFAULT_LENGTH = 32
+
+    # Logging the user regardless course exits or not otherwise Django will show default login page
+    try:
+        user_obj = User.objects.get(username=username)
+    except User.DoesNotExist:
+        password = ''.join(random.sample(string.ascii_letters, RANDOM_PASSWORD_DEFAULT_LENGTH))
+        user_obj = User.objects.create_user(username=username, email=email, password=password, first_name=first_name, last_name=last_name)
+    user_obj.backend = 'django.contrib.auth.backends.ModelBackend'
+    django.contrib.auth.login(request, user_obj)
+
+    is_instructor = check_if_instructor(roles, username, course_id)
+
+    try:
+        course_details = Course.objects.get(canvas_id=course_id)
+    except ObjectDoesNotExist:
+        if is_instructor:
+            canvas_long_id = canvas_id_to_incremented_id(course_id)
+            course_details = Course.objects.create(id=canvas_long_id, canvas_id=course_id, name=course_name)
+            CourseViewOption.objects.create(course_id=canvas_long_id)
+
+    if course_details.term_id is None:
+        logger.info(f"Course {course_id} don't have a cron run yet")
+    lti_globals = {
+        "username": username,
+        "launch_id": launch_id,
+        "user_courses_info": {"course_id": course_id, "course_name": course_name},
+        "is_superuser": user_obj.is_superuser,
+        "login": settings.LOGIN_URL,
+        "logout": settings.LOGOUT_URL,
+        "user_image": user_img,
+        "primary_ui_color": settings.PRIMARY_UI_COLOR,
+        "help_url": settings.HELP_URL,
+        "google_analytics_id": settings.GA_ID
+    }
+    return lti_globals
+
+
+@csrf_exempt
+def login(request):
+    logger.info("LTI Login called...........")
+    tool_conf = get_tool_conf()
+    launch_data_storage = DjangoCacheDataStorage(cache_name='default')
+    oidc_login = DjangoOIDCLogin(request, tool_conf, launch_data_storage=launch_data_storage)
+    target_link_uri = get_launch_url(request)
+    return oidc_login.enable_check_cookies().redirect(target_link_uri)
+
+
+@require_POST
+@method_decorator(xframe_options_exempt, name='dispatch')
+@method_decorator(csrf_exempt, name='dispatch')
+def launch(request):
+    logger.info("LTI Launch of the tool")
+    tool_conf = get_tool_conf()
+    launch_data_storage = DjangoCacheDataStorage(cache_name='default')
+    message_launch = DjangoMessageLaunch(request, tool_conf, launch_data_storage=launch_data_storage)
+    lti_globals = validate_lti_1_3_request(request, message_launch)
+    template = loader.get_template('frontend/index.html')
+    context = {
+        'lti_globals': lti_globals
+    }
+    # return redirect(url)
+    logger.info({'lti_globals': lti_globals })
+    return HttpResponse(template.render(context, request))
```

Comment:
Instead of redirect I opted for rending the `index.html` and I am passing all the launch related info all the globals such as username, user_image url etc.

Up until now context_processors.get_myla_globals ` was sending that info. But now that is not needed anymore